docs/configuration.md

```diff
@@ -304,6 +304,16 @@ validation:
         - ^https?://([^/]+\.)*example\.com/
       deny:
         - ^https?://www\.example\.com/
+    configMediaTypes:
+      allow:
+        - ^.*$
+      deny:
+        - ^application\/vnd\.bad\.config\.v1\+json$
```

Comment:
Somewhat related to my other comment, but slightly curious if there's _practical_ examples that require a regex, or if these could be addressed by taking a list of accepted mediatypes. 

Regexes can be powerful, but also can cause overhead, and a footgun (easy to get them wrong, e.g. someone putting `application/vnd.oci.image.layer.v1.tar`, and not ending up with what they expect). If regexes are _really_ needed, perhaps there should be a validation that regexes _must_ have proper start/end markers (start with `^`, end with `$`). Alternatively, these could be literal lists (for now), and support for regexes could be added later (in which case `^` and `$` could be used to switch between regex or plain string)

I guess these allow/deny lists would allow;

1. disabling support for default (but legacy) manifest types
2. adding new mediatypes that are not (yet) part of the OCI spec (in which case the exact mediaTypes would likely be known)
3. make handling of the "unknown" mediaTypes more restrictive, as it's currently accepting "anything";

https://github.com/distribution/distribution/blob/9329f6a62b67d5e06d50dc93997c7705a075fcd9/registry/storage/ocimanifesthandler.go#L123-L126

- For `1.`, the list of "default" mediaTypes is known, so a non-regex option could be suitable
- For `2.` (as mentioned) the exact mediaTypes would likely be known
- `3.` is the "interesting" one, as the current default is quite permissive;
    - because there's currently no known list, a `deny` list may not be very useful; "anything" could already be uploaded, and denying a specific type would be of limited use.
    - an allow-list could make sense (if it's known what types to support), but wondering about the practicality of a regular expression for this other than a "slightly" less DRY configuration, such as `^foo(bar|baz)$` - but even that likely only applies to a "limited" set of known (`bar|baz`) options. The moment any "wildcard" match is part of the RegEx, things are basically back to "accept anything".